pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -189,6 +192,36 @@ func AllNodesForNodeClaim(ctx context.Context, c client.Client, nodeClaim *v1bet
 	return lo.ToSlicePtr(nodeList.Items), nil
 }
 
+// NodePoolForNodeClaim is a helper function that takes a v1beta1.NodeClaim and finds its matching NodePool using its ownerRefs
+func NodePoolForNodeClaim(ctx context.Context, c client.Client, nodeClaim *v1beta1.NodeClaim) (*v1beta1.NodePool, error) {
```

Comment:
IMO we should probably handle the error logging at the call site rather than within a helper function.